File: python/dictionary.py
import requests
import json
import sys

def lookup(word):
    response = requests.get(f"https://www.dictionaryapi.com/api/v3/references/collegiate/json/{word}?key=<API>")
    result = json.loads(response.content)[0]
    definitions = "\n".join(result["shortdef"])

    print("English-English result:")
    print(f"{word}: {definitions}")

    print("\nChinese result:")
    print("<insert Chinese translation here>")

if __name__ == "__main__":
    lookup(sys.argv[1])

File: ruby/dictionary.rb
require 'rest-client'
require 'json'

def lookup(word)
  response = RestClient.get "https://www.dictionaryapi.com/api/v3/references/collegiate/json/#{word}?key=<your_api_key_here>"
  result = JSON.parse(response.body)[0]
  definitions = result["shortdef"].join("\n")

  puts "English-English result:"
  puts "#{word}: #{definitions}"

  puts "\nChinese result:"
  puts "<insert Chinese translation here>"
end

lookup(ARGV[0])
